blues/airnote.live
# feat: date range dropdown

# Problem Context

The current Grafana powered Airnote dashboards have ElasticDB on the backend, meaning they can pull back data from various user defined time frames with incredible speed, our Airnote dashboard doesn't have that same database architecture at its disposal, but we should still give users the option to choose different time frames.

## Changes

Added a dropdown to the dashboard's charts section of the app so users can choose to view historical readings of the last day, 2 days, 3 days, 5 days, 7 days, and update the Historical Readings header to reflect those changes.

Reconfigured the server to take in an extra, optional argument to fetch different amounts of data for the charts.

Beefed up the e2e tests to ensure the dropdown works correctly.

## Screenshot (if applicable)

https://user-images.githubusercontent.com/20400845/185434900-cdd00892-2fa1-4593-914d-7abdd7a4a484.mov

## Testing

Unit / integration / e2e tests?

Updated e2es to test the date range picker

Steps to test manually?

1. Go to http://localhost:5555/dev:864475044215258/dashboard
2. Scroll down to the charts
3. See the date picker and note it defaults to 7 days (just like the AQI Average history component above it)
4. Select another date from the picker and see the charts update, while the AQI average above remains unchanged (this is expected behavior)

Any other sorts of testing notes to include?

A "staging version" of the site is available here: https://airnote-live-pr-45.onrender.com/dev:864475044215258/dashboard

## Any other related PRs

n/a

## Ticket(s)

https://trello.com/c/KQDfJymP/88-add-airnote-dropdown-to-change-dates-of-data-pulled-back-and-displayed
